voltdb/src/main/java/com/yahoo/ycsb/db/voltdb/VoltClient4.java

```diff
@@ -0,0 +1,292 @@
+ * LICENSE file.
+ */
+
+/*
+ * This client provideds a wrapper layer for running the Yahoo Cloud Serving
+ * Benchmark (YCSB) against VoltDB. This benchmark runs a synchronous client
+ * with a mix of the operations provided below. YCSB is open-source, and may
+ * be found at https://github.com/brianfrankcooper/YCSB. The YCSB jar must be
+ * in your classpath to compile this client.
+ */
+package com.yahoo.ycsb.db.voltdb;
+
+import java.nio.ByteBuffer;
+import java.nio.charset.Charset;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+import java.util.Vector;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.voltdb.VoltTable;
+import org.voltdb.client.Client;
+import org.voltdb.client.ClientResponse;
+import org.voltdb.client.ClientResponseWithPartitionKey;
+import org.voltdb.client.NoConnectionsException;
+
+import com.yahoo.ycsb.ByteArrayByteIterator;
+import com.yahoo.ycsb.ByteIterator;
+import com.yahoo.ycsb.DB;
+import com.yahoo.ycsb.DBException;
+import com.yahoo.ycsb.Status;
+import com.yahoo.ycsb.db.voltdb.sortedvolttable.VoltDBTableSortedMergeWrangler;
+
+/**
+ * A client that can be used by YCSB to work with VoltDB.
+ */
+public class VoltClient4 extends DB {
+
+  private Client mclient;
+  private byte[] mworkingData;
+  private ByteBuffer mwriteBuf;
+  private boolean useScanAll = false;
+  private static final Charset UTF8 = Charset.forName("UTF-8");
+  
+  private Logger logger = LoggerFactory.getLogger(VoltClient4.class);
+  
+  private YCSBSchemaBuilder ysb =  null;
+
+  @Override
+  public void init() throws DBException {
+    Properties props = getProperties();
+    String servers = props.getProperty("voltdb.servers", "localhost");
+    String user = props.getProperty("voltdb.user", "");
+    String password = props.getProperty("voltdb.password", "");
+    String strLimit = props.getProperty("voltdb.ratelimit");
+    String useScanAllParam = props.getProperty("voltdb.scanall", "no");
+
+    if (useScanAllParam.equalsIgnoreCase("YES")) {
+      useScanAll = true;
+    }
+
+    int ratelimit = strLimit != null ? Integer.parseInt(strLimit) : Integer.MAX_VALUE;
+    try {
+      mclient = ConnectionHelper.createConnection(Thread.currentThread().getId(), servers, user, password, ratelimit);
+      
+      ysb = YCSBSchemaBuilder.getInstance(mclient);
+      ysb.loadClassesAndDDLIfNeeded();
+
+    } catch (Exception e) {
+      logger.error("Error while creating connection: ", e);
+      throw new DBException(e.getMessage());
+    }
+    mworkingData = new byte[1024 * 1024];
+    mwriteBuf = ByteBuffer.wrap(mworkingData);
+  }
+
+  /**
+   * @return true if we have a live DB connection
+   */
+  public boolean hasConnection() {
+
+    if (mclient != null && mclient.getConnectedHostList().size() > 0) {
+      return true;
+    }
+
+    return false;
+
+  }
+
+  @Override
+  public void cleanup() throws DBException {
+    
+    try {
+      mclient.drain();
```

Comment:
As a nit, should have a null guard here.

Reply:
Fixed.